cypress/integration/channel/message_draft_then_switch_channel_spec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Message Draft and Switch Channels', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('M14358 Message Draft Pencil Icon Visible in Channel Switcher', () => {
+        // 1. In a test channel, type some text in the message input box
+        // 2. Do not send the post
+        cy.get('#sidebarItem_town-square').scrollIntoView();
+        cy.get('#sidebarItem_town-square').should('be.visible').click();
+
+        // * Validate if the channel has been opened
+        cy.url().should('include', '/ad-1/channels/town-square');
+
+        // * Validate if the draft icon is not visible on the sidebar before making a draft
+        cy.get('#sidebarItem_town-square').scrollIntoView();
+        cy.get('#sidebarItem_town-square #draftIcon').should('be.not.visible');
+
+        // Type in some text into the text area of the opened channel
+        cy.get('#post_textbox').type('whalecome to the jungle');
+
+        // 3. Switch to another channel
+        cy.get('#sidebarItem_veritatis').scrollIntoView();
+        cy.get('#sidebarItem_veritatis').should('be.visible').click();
+
+        //* Validate if the newly navigated channel is open
+        cy.url().should('include', '/ad-1/channels/veritatis');
+
+        // 4. Press CTRL/CMD+K (RN apps "Jump To")
+        cy.typeCmdOrCtrl().type('K', {release: true});
+
+        // 5. Type the first few letters of the channel name you typed the message draft in
+        cy.get('#quickSwitchInput').type('tow');
+        cy.wait(500);  // eslint-disable-line
+
+        // * Suggestion list should be visible
+        cy.get('#suggestionList').should('be.visible');
+
+        //* Validate if the draft icon is visible to left of the channel name in the filtered list
+        cy.get('#suggestionList_town-square #draftIcon').should('be.visible');
```

Comment:
There's no such element ID that generates `#suggestionList_town-square`.  Did you happen to add it on separate file?  If not, I suggests to add it but as ``id={`switchChannel_${channel.name}`}`` in [switch_channel_provider](https://github.com/mattermost/mattermost-webapp/blob/1deda37f692d4dc6627ec522d76a3c328d3476ff/components/suggestion/switch_channel_provider.jsx#L130). Then you may use it as,
```suggestion
        cy.get('#switchChannel_town-square #draftIcon').should('be.visible');
```